binary-option/program/src/state.rs

```diff
@@ -0,0 +1,42 @@
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult, program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+use crate::error::BinaryOptionError;
+use borsh::{BorshDeserialize, BorshSerialize};
+
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone)]
+pub struct BinaryOption {
+    pub decimals: u8,
+    pub circulation: u64,
+    pub settled: bool,
+    pub escrow_mint_account_pubkey: Pubkey,
+    pub escrow_account_pubkey: Pubkey,
+    pub long_mint_account_pubkey: Pubkey,
+    pub short_mint_account_pubkey: Pubkey,
+    pub owner: Pubkey,
+    pub winning_side_pubkey: Pubkey,
```

Comment:
it's just jumbled bytes, could make this change but didn't want to deal with optional types in Borsh. Client/Protocol ease of use tradeoff

Reply:
IMO easies to just check settled bit and ignore `winning_side_pubkey` if `settled == 0`. This makes deserializing in Python easier, because I don't have to think about optional types

Reply:
I'm suggesting to use an enum, not an option here.  That should be pretty easy to deserialize, right?

Reply:
ah yeah that's also fine, but I think it's pretty minor. The binary option struct is only 202 bytes, and I don't think you get _that_ much performance gain from this change (interface is abstracted away from user as well). Always happy to refactor if and when it becomes necessary, but this works as of now, so I don't see a huge need to fix this in particular

Reply:
It's more a question of readability and ease of comprehension for these two variables.  An enum is clearer than a bool + separate default-initialized pubkey.